<launch>
    <arg name="output_topic" default="nn_viz/output" />
    <arg name="model_name" default="lines3_traced_forward.pth"/>

    <node name="$(anon nn_viz)" pkg="igvc_perception" type="nn_viz.py" output="screen">
        <!-- These sizes must match the model input dimensions. -->
        <param name="image_resize_width" value="400" />
        <param name="image_resize_height" value="400" />

    <!--     <param name="model_path" value="$(find igvc_sandbox)/models/IGVCModel_135.pt" /> -->
    <!--     <param name="model_path" value="$(find igvc_sandbox)/models/SE_Resnet50_trace.pth" /> -->
        <param name="model_path" value="$(find igvc_sandbox)/models/$(arg model_name)" />

        <param name="force_cpu" value="false" /> <!-- Use cpu instead of GPU (will run very slowly). -->

        <param name="output_topic" value="$(arg output_topic)" />
        <param name="input_topic" value="/usb_cam_center/image_raw/compressed" />
    </node>
</launch>
